tests/e2e/scripts/helm_keadm_e2e.sh

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 The KubeEdge Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+KUBEEDGE_ROOT=$PWD
+WORKDIR=$(dirname $0)
+HELM_DIR=$KUBEEDGE_ROOT/build/helm
+E2E_DIR=$(realpath $(dirname $0)/..)
+IMAGE_TAG=$(date +%Y%m%d)
+
+function cleanup() {
+  sudo pkill edgecore || true
+  helm uninstall cloudcore -n kubeedge && kubectl delete ns kubeedge  || true
+  kind delete cluster --name test
+  sudo rm -rf /var/log/kubeedge /etc/kubeedge /etc/systemd/system/edgecore.service $E2E_DIR/keadm/keadm.test $E2E_DIR/config.json
+}
+
+function build_keadm() {
+  make all WHAT=keadm
+  cd $E2E_DIR
+  ginkgo build -r keadm/
+}
+
+function prepare_cluster() {
+  kind create cluster --name test
+
+  echo "wait the control-plane ready..."
+  kubectl wait --for=condition=Ready node/test-control-plane --timeout=60s
+
+  kubectl create clusterrolebinding system:anonymous --clusterrole=cluster-admin --user=system:anonymous
+
+  # edge side don't support kind cni now, delete kind cni plugin for workaround
+  # kubectl delete daemonset kindnet -nkube-system
+}
+
+function build_cloud_image() {
+  cd $KUBEEDGE_ROOT
+  make cloudimage -f $KUBEEDGE_ROOT/Makefile -e IMAGE_TAG=$IMAGE_TAG
+  kind load docker-image kubeedge/cloudcore:$IMAGE_TAG --name test
+}
+
+function start_kubeedge() {
+  sudo mkdir -p /var/lib/kubeedge
+  cd $KUBEEDGE_ROOT
+  export KUBECONFIG=$HOME/.kube/config
+
+  cd $HELM_DIR
+  SET_ARGS="--set cloudCore.modules.cloudHub.advertiseAddress[0]=$MASTER_IP --set cloudCore.image.tag=$IMAGE_TAG --set cloudCore.service.enable=false"
+  helm upgrade --install --wait --timeout 30s cloudcore ./cloudcore --namespace kubeedge --create-namespace -f ./cloudcore/values.yaml $SET_ARGS
+  export MASTER_IP=`kubectl get node test-control-plane -o jsonpath={.status.addresses[0].address}`
+
+  # ensure tokensecret is generated
+  while true; do
+      sleep 3
+      kubectl get secret -nkubeedge 2>/dev/null | grep -q tokensecret && break
+  done
+  
+  cd $KUBEEDGE_ROOT
+  export TOKEN=$(sudo _output/local/bin/keadm gettoken --kube-config=$KUBECONFIG)
+  sudo systemctl set-environment CHECK_EDGECORE_ENVIRONMENT="false"
+  sudo -E CHECK_EDGECORE_ENVIRONMENT="false" _output/local/bin/keadm join --token=$TOKEN --cloudcore-ipport=$MASTER_IP:10000 --edgenode-name=edge-node
```

Comment:
this join command will use the latest version release as edgecore, but not the latest codes. we'd better build a new release with the latest codes and use --kubeedge-version flag to specify which new release we will use.

Reply:
We can just build a kubeedge release package and please it to /etc/kubeedge now..  but `/etc/kubeedge` is not a good place for the package.

Reply:
Currently, we only test the hostnetwork case, maybe need to test the external iptablesmgr, any ideas?